src/Driver/API/SQLite/UserDefinedFunctions.php

```diff
@@ -11,6 +15,25 @@
  */
 final class UserDefinedFunctions
 {
+    private const DEFAULT_FUNCTIONS = [
+        'sqrt' => ['callback' => [SqlitePlatform::class, 'udfSqrt'], 'numArgs' => 1],
+        'mod'  => ['callback' => [SqlitePlatform::class, 'udfMod'], 'numArgs' => 2],
+        'locate'  => ['callback' => [SqlitePlatform::class, 'udfLocate'], 'numArgs' => -1],
+    ];
+
+    /**
+     * @param Closure(string, callable, int): bool                   $callback
+     * @param array<string, array{callback: callable, numArgs: int}> $additionalFunctions
+     */
+    public static function register(Closure $callback, array $additionalFunctions): void
+    {
+        $userDefinedFunctions = array_merge(self::DEFAULT_FUNCTIONS, $additionalFunctions);
```

Comment:
I'm curious if we should deprecate additional functions. The DBAL uses the default ones to achieve parity with the rest of the platforms. Any new functions wouldn't be part of the abstraction. Users should be able to register their functions themselves via the native connection.

Reply:
> Users should be able to register their functions themselves via the native connection.

I think that's a good idea. It should be pretty straightforward to build a middleware that adds custom functions on `Driver::connect()`.

Reply:
Here you go: #5742.